src/redis_lock/__init__.py

```diff
@@ -318,9 +324,11 @@ def _stop_lock_renewer(self):
         logger_for_refresh_exit.debug("Renewal thread for Lock(%r) exited.", self._name)
 
     def __enter__(self):
-        acquired = self.acquire(blocking=True)
-        if not acquired:
+        acquired = self.acquire(blocking=self.blocking)
+        if not acquired and self.blocking:
```

Comment:
Looks like you could use a subbranch here so we don't needlessly check for `not acquired` two times.

Reply:
Hi @ionelmc Something else should be done here? Or will it wait till you have the opportunity to merge?